command/agent/command.go

```diff
@@ -1081,8 +1081,12 @@ func (c *Command) handleReload(config *Config) (*Config, error) {
 		errs = multierror.Append(errs, fmt.Errorf("Failed unloading checks: %s", err))
 		return nil, errs
 	}
+	if err := c.agent.unloadMetadata(); err != nil {
```

Comment:
Oh wait - I thought that's what the snapshot was for, but it's just for checks. Reload depends on the config read finding the errors before it starts changing state but it doesn't roll back once it starts to apply the changes. We should have `readConfig()` validate the metadata so it's caught early.